### DISCUSSION TITLE
Using a monorepo with several Next projects

### DISCUSSION THREAD

**Discussion by Nick-Mazuk** [vercel/next.js#19224] [Help]
I'm about to create a new repo with multiple NextJS sites. In production, each site will be on the same domain but have a unique [base path](https://nextjs.org/docs/api-reference/next.config.js/basepath) and use [Vercel's rewrites](https://vercel.com/docs/configuration#project/rewrites).

Is it possible to run each of these NextJS installations on localhost at the same time? That way in development I can navigate the different sites as if it's in production?

Are there any best practices for doing something like this?

**Comment by leerob** [comment]
Yes! You can use Vercel's monorepo support: https://vercel.com/blog/monorepos. 

That will allow you to deploy multiple Next.js apps from the same repo. We're publishing a blog post very soon on Next.js incremental adoption strategies that will include this!

**Reply by leerob** [reply]
More info on incremental adoption: https://github.com/vercel/next.js/pull/19226

**Reply by Nick-Mazuk** [reply]
Yes, that's exactly what I was hoping to do for production. I guess more specifically I was hoping to be able to run each of the apps at the same time on localhost as well. I probably didn't make that clear in my original question, sorry. I edited the original post so hopefully it's clearer.

**Reply by leerob** [reply]
`cd` into each folder and run each with a different port, e.g. `next -p 4000`. Could write a script in your `package.json` to do this 👍

**Reply by mikematos84** [reply]
@leerob 

Would this be possible with NextJS apps on different versions? I have a monorepo I am working in and I have a need to have one NextJS app on 14 using the pages router and one on 15 using the apps router for testing how to integrate our UI library into different versions. 

Since Next 14 uses React 18 and Next 15 uses React 19, I am running into issues building since it has trouble targeting the right react version even if I use "overrides" or "resolutions" within their package.json files
